src/foundry/foundry.js/applications/formApplications/defaultTokenConfig.d.ts

```diff
@@ -3,21 +3,21 @@ declare global {
    * A sheet that alters the values of the default Token configuration used when new Token documents are created.
    */
   class DefaultTokenConfig<
-    Options extends FormApplication.Options = FormApplication.Options
-  > extends FormApplication<Options> {
+    Options extends FormApplication.Options = FormApplication.Options,
+    Data extends DefaultTokenConfig.Data = DefaultTokenConfig.Data
+  > extends FormApplication<Options, Data, foundry.data.TokenData['toObject']> {
     constructor(object: unknown, options: Options);
 
     data: foundry.data.TokenData;
 
-    object: foundry.data.TokenData['toObject'];
+    override object: foundry.data.TokenData['toObject'];
```

Comment:
this is not correct, I guess you mean the return type. I would just use `foundry.data.TokenData['_source']` instead, that's the correct type. Also, do we even need this here? I know it's being set explicitly but if we also use this type as the generic parameter for `FormApplication`, we shouldn't need to override it here.